### PR TITLE
docs: update current release processes

### PR DESCRIPTION
**Pull request by zalberico**
### Description

Resolves: https://github.com/urbit/urbit/issues/6198

Describes the new release process in maintainers and new requirements around issues, and pull requests for contributing. I also pulled out a lot of the Vere specifics now that we're going to be shipping Vere from its new repo.

I tried to match the style and requirements in the new Vere repo. Based on feedback from @jalehman I explained what the pre-release moons are more generally so others in the Urbit development ecosystem can understand them and use them (assuming the risks that come along with that).

There was a lot here to refactor and will likely be things I have to change - thanks for taking a look.

### Related
Vere: https://github.com/urbit/vere

### CODE REVIEW

File: CONTRIBUTING.md
Comment (Fang-) [inline CONTRIBUTING.md:99]: Potentially worth adding here that it's highly desirable for every individual commit to at least be in a compiling & run-able state. Obviously-broken "wip" commits should be discouraged.

File: CONTRIBUTING.md
Comment (Fang-) [inline CONTRIBUTING.md:113]: This header is superfluous, since you're already writing into the PR description field.

With that, only the "Related" header remains, which seems optional to me, and might also be served by a more modest "See also:" at the start of a new paragraph.

I'd consider putting "resolves #n" at the bottom instead of leading with it, since it's mostly administrative, but this seems more like a personal preference thing.

File: CONTRIBUTING.md
Comment (Fang-) [inline CONTRIBUTING.md:130]: Doesn't this only happen once it (eventually) gets merged into master?

File: CONTRIBUTING.md
Comment (Fang-) [inline CONTRIBUTING.md:143]: fwiw I have never heard anyone say this. It's usually just "make/update a pill".

File: CONTRIBUTING.md
Comment (Fang-) [inline CONTRIBUTING.md:185]: This seems duplicate with the sections above.

File: CONTRIBUTING.md
Comment (Fang-) [inline CONTRIBUTING.md:198]: This feels outdated. I haven't needed to be concerned about this in a long time.

Comment (Fang-) [reply CONTRIBUTING.md:198]: We _may_ want to mention the ability to build pills off non-base desks though, somewher in the section above.

If, during development, you don't want to bother writing state migration logic yet, you can `|new-desk %new-base`, mount that, put the new files in there, then `+solid %new-base` to build the pill off that desk.

File: CONTRIBUTING.md
Comment (Fang-) [inline CONTRIBUTING.md:229]: We'll stop using brass pills for mainnet boot Soon™. (;

File: CONTRIBUTING.md
Comment (Fang-) [inline CONTRIBUTING.md:224]: Wondering if we should just tell people to use `/bin/solid.pill` from this repo in the default instructions. If they're making arvo contributions, they're checking out the repo anyway, and the develop branch may have sys changes that they'll want to have loaded in. (Although, even then, the pill isn't hard-guaranteed to be up to date, they may need to copy in /pkg/arvo/sys and/or other files manually regardless...)

File: CONTRIBUTING.md
Comment (Fang-) [inline CONTRIBUTING.md:251]: Somewhat duplicate with instructions above.

File: CONTRIBUTING.md
Comment (Fang-) [inline CONTRIBUTING.md:262]:
```suggestion
[^1]: Unless you specified the pier name using the `-c` flag.
```

File: CONTRIBUTING.md
Comment (Fang-) [inline CONTRIBUTING.md:41]: As mentioned out of band, not a huge fan of this. I could be convinced otherwise, but presently unclear what this gains us over "post the branch name to the issue when work has started", while adding the cost of more difficult to remember/work with branch names. Do we really care that much about having branch->issue lookup?

File: MAINTAINERS.md
Comment (Fang-) [inline MAINTAINERS.md:21]: I thought we were gonna keep this secret to prevent people getting themselves into trouble?

File: MAINTAINERS.md
Comment (Fang-) [inline MAINTAINERS.md:28]: Small suggestion, we could incorporate the not-a-table of what gets deployed where into the table.

```suggestion
-----------------------------------------------------------------------------------
 Branch     |            Moon          | Target audience     | Contains
-----------------------------------------------------------------------------------
`develop`   | `~binnec-dozzod-marzod`  | Kernel developers   | Latest `develop` branch commit
`release`   | `~marnec-dozzod-marzod`  | Early Adopters      | Latest `release` branch commit
`release`   | `~doznec-dozzod-marzod`  | App Developers      | Latest release candidate
`master`    | `~zod`                   | Everyone else       | Latest release
```

File: MAINTAINERS.md
Comment (Fang-) [inline MAINTAINERS.md:45]: We could just always do this, for consistency sake. If they haven't diverged it would just be a no-op right? If not, maybe we can just automate the check/merge?

Notably this means develop may be "behind" release at certain points in time. Feels vaguely dangerous. I think the previous flow of always deploying to binnec first, and pulling what's there directly onto marnec when it's considered good avoids this, at the cost of keeping binnec "occupied" until a release has gone out. I'm assuming that's why we're doing the above instead?

File: MAINTAINERS.md
Comment (Fang-) [inline MAINTAINERS.md:107]: (Very nitpicky, but) a tiny bit surprised this work, the below looks more natural to me.

```suggestion
git shortlog --no-merges LAST_RELEASE..
```

File: README.md
Comment (Fang-) [inline README.md:13]: Wondering if we should specify with a bit more detail, like "the software management UI", or "the homepage UI". Although both of those are still too narrow, because it includes _backend_ software, for notifications, S3 settings, etc.

Also, are we _really_ doing "garden" as the repo name? Have we not suffered enough?? urbit/landscape is archived, we could obliterate it and replace it with ""garden"".

**Review by Fang- — COMMENTED** [review]
Bunch of comments, both process-level and document-level.

File: CONTRIBUTING.md
Comment (zalberico) [reply CONTRIBUTING.md:113]: +1 on headers, I think link at the top is good though

File: CONTRIBUTING.md
Comment (zalberico) [reply CONTRIBUTING.md:130]: As part of this I'd set the default branch to develop - I think then it closes as expected

File: CONTRIBUTING.md
Comment (zalberico) [reply CONTRIBUTING.md:143]: +1 this was old text, I'll remove it and use more simple/normal language

File: CONTRIBUTING.md
Comment (zalberico) [reply CONTRIBUTING.md:41]: I think this is actually important. It does two things, it forces an issue to exist in order to adhere to the branch naming spec and it makes it obvious what the branch is for that issue (can tab complete). CC @mcevoypeter who came up with the idea.

File: MAINTAINERS.md
Comment (zalberico) [reply MAINTAINERS.md:21]: Yeah, this was important to Josh. I should put a loud warning here though.

WARNING: Using early pre-release moons (binnec and marnec) is high risk and may require you to breach if there's an issue.

Something like that.

File: README.md
Comment (zalberico) [reply README.md:13]: Yeah, I can do that. We have the ok to move garden from the urbit org to tloncorp now, I can rename it to landscape as part of that.

File: MAINTAINERS.md
Comment (jalehman) [inline MAINTAINERS.md:22]: I'd think app developers should receive releases earlier in the cycle than early adopters. I'd consider them more likely to uncover issues since they interface directly with kernel APIs.

File: MAINTAINERS.md
Comment (jalehman) [reply MAINTAINERS.md:21]: It's a calculated benefit. We have more qualified eyeballs to make bugs shallow than has been assumed to date.

WARNING: If you lack the requisite skills to troubleshoot and fix kernel issues, you should not sync from `develop/~binnec`. If you're not prepared to breach your ship in response to an issue stemming from an early release, do not use these pre-release moons.

File: README.md
Comment (jalehman) [reply README.md:13]: I'd remove this one entirely from the list, since it's not part of "Urbit" in the same way that these others are.

File: README.md
Comment (jalehman) [inline README.md:20]:
```suggestion
[urbit.org/getting-started][start]. You'll be on the live network in a
```

File: README.md
Comment (jalehman) [inline README.md:45]: remove

File: README.md
Comment (jalehman) [inline README.md:23]:
```suggestion
[start]: https://urbit.org/getting-started/cli/
```

File: CONTRIBUTING.md
Comment (zalberico) [reply CONTRIBUTING.md:113]: One nice thing about the header is I think I can set defaults for text fields. Makes it more obvious what's required (and the rendered headers look nice in the field, like in this PR for example).

File: CONTRIBUTING.md
Comment (Fang-) [reply CONTRIBUTING.md:130]: That feels like it might be closing too eagerly though. Mainnet people would still be experiencing the issue, naively search for it and find nothing, even though it's hidden away in closed issues. Maybe not a big deal, but worth taking into account.

File: CONTRIBUTING.md
Comment (Fang-) [reply CONTRIBUTING.md:113]: fwiw I personally always bounce super hard on pre-filled PR/issue description fields. Something about them feels, idk, belittling maybe?

Ours would be very minimal, so it's not as bad, but also people should be reading the contributing.md anyway. iirc Github shows a "have your read the contributing.md" for first-time submitters, maybe we can put that in the input field instead? ...But perhaps that's worse. (^:

File: README.md
Comment (sigilante) [inline README.md:13]: There's also a question of whether to redirect to Groups/`tloncorp` repos here as well, particularly given the equivocation around the name "Landscape" over the years.

Comment (sigilante) [reply README.md:13]: And I assume we should make reciprocal links from those...

**Comment by sigilante** [comment]
I'd also like to know more about CI/unit testing/regression testing procedures.

File: CONTRIBUTING.md
Comment (zalberico) [reply CONTRIBUTING.md:130]: I think it's a setting, so we don't need it to automatically close the issue.

File: CONTRIBUTING.md
Comment (zalberico) [reply CONTRIBUTING.md:198]: I'm going to say the latter is out of scope (this doc is already probably too in the weeds)

File: README.md
Comment (zalberico) [reply README.md:13]: Yeah - Josh asked to just remove those so I did

**Comment by zalberico** [comment]
Thanks for all of the helpful feedback. I've gone through and updated all of the docs based on it and they're in a much better place than they were.

When we merge this, we'll want to pair that with the following things too:
- Changing the name of `next/arvo` to `develop`
- Changing the default branch in the repo from `master` to `develop`

**Comment by zalberico** [comment]
> I'd also like to know more about CI/unit testing/regression testing procedures.

Yeah me too, but I'm probably not the person to write that (or at least would need to learn more details before I could). I'd like to have that be part 2 and not block merging this refactor.

**Review by jalehman — APPROVED** [review]

**Pull request merged**
